fastchat/serve/model_worker.py

```diff
@@ -48,6 +48,13 @@ def load_model(model_path, num_gpus):
         }
 
     tokenizer = AutoTokenizer.from_pretrained(model_path)
+    if isinstance(tokenizer, LlamaTokenizer):
+        # This is a hack to make the new transformers' tokenizer work with the vicuna model
+        # trained with the transformers before commit: c0f99b4
+        # TODO: remove this when we retrain the vicuna model
+        tokenizer.bos_token_id = tokenizer.sp_model.bos_id()
+        tokenizer.eos_token_id = tokenizer.sp_model.eos_id()
```

Comment:
I think fixing the `tokenizer_config.json` should be enough to remove this? 
```python 
{
  "add_bos_token": true,
  "add_eos_token": false,
  "bos_token": {
    "__type": "AddedToken",
    "content": "<s>",
    "lstrip": false,
    "normalized": true,
    "rstrip": false,
    "single_word": false
  },
  "clean_up_tokenization_spaces": false,
  "eos_token": {
    "__type": "AddedToken",
    "content": "</s>",
    "lstrip": false,
    "normalized": true,
    "rstrip": false,
    "single_word": false
  },
  "model_max_length": 1000000000000000019884624838656,
  "pad_token": null,
  "sp_model_kwargs": {},
  "tokenizer_class": "LlamaTokenizer",
  "unk_token": {
    "__type": "AddedToken",
    "content": "<unk>",
    "lstrip": false,
    "normalized": true,
    "rstrip": false,
    "single_word": false
  }
}
```
and the special map:
```python 
{
  "bos_token": {
    "content": "<s>",
    "lstrip": false,
    "normalized": true,
    "rstrip": false,
    "single_word": false
  },
  "eos_token": {
    "content": "</s>",
    "lstrip": false,
    "normalized": true,
    "rstrip": false,
    "single_word": false
  },
  "unk_token": {
    "content": "<unk>",
    "lstrip": false,
    "normalized": true,
    "rstrip": false,
    "single_word": false
  }
}
```
Sorry if this doesn't work!

Reply:
Thanks for the suggestion @ArthurZucker! I just tested it, and it seems to be working with our model weights! We may want to update our public delta weights to make it support the latest huggingface, cc'ing @merrymercy.

Reply:
Thanks for teaching us this. The weights have been updated https://huggingface.co/lmsys/vicuna-13b-delta-v0/commit/7976eed3121f3416b5745b8ecbdc6a10eb629f1e